### DISCUSSION TITLE
WSL + ccache

### DISCUSSION THREAD

**Discussion by Conan179** [Freetz-NG/freetz-ng#835] [Off-topic]
Ähm mal doof gefragt, wie mache ich einen bug report? Ich wollte für meine 6591 ein freetz erstellen, schlägt aber fehl.

**Comment by fda77** [comment]
"als würde der der es liest durch deine augen schauen"

 - https://www.chiark.greenend.org.uk/~sgtatham/bugs.html
 - https://www.chiark.greenend.org.uk/~sgtatham/bugs-de.html

Da es bei freetz aber niemanden gibt der dazu viel lust hat wäre optimal: Fehler bemerken - beheben - pullrequest erstellen

**Reply by Conan179** [reply]
Den "fehler" habe ich inzwischen gefunden ich musste in meinem wsl "mkdir /run/user/1000/ccache-tmp" und "sudo chmod 777 /run/user/1000/ccache-tmp".
Ich geh davon aus das es wsl spezifisch ist und nicht mit freetz-ng zu tun hat.

**Reply by jwgn** [reply]
Danke, damit hat es nun bei mir auch endlich funktioniert.

`7490_07.57.all_freetz-ng-22364-945a29e28` erfolgreich unter WSL kompiliert und geflasht.

Wie bist du auf die Lösung gekommen? Sollte das eventuell in die Doku?

**Reply by Conan179** [reply]
@jwgn kurz gesagt selbst drauf gekommen. Ich hab mir die fehler meldung angeschaut und hab dann quick&dirty "mkdir /run/user/1000/ccache-tmp" und "sudo chmod 777 /run/user/1000/ccache-tmp" gemacht un dann lief es. Das WSL nutzte ich momentan ausschlich für Freetz, wens zerschossen gewesen wäre, hätte ich es schlicht neu innstalliert.

**Comment by fda77** [comment]
Mit WSL hatte ich noch nie zu tun. Muss man da irgend etwas besonderes machen? Ansonsten, funktioniert ein so ein erstelltes Image?

**Reply by Conan179** [reply]
Eigentlich nicht, prerequisites install ubuntu22 (in meinem Fall) und dann freetz aus gecheckt nach anleitung.

**Comment by fda77** [comment]
* efd3c6429 ccache does not work with wsl

wenn ccache im menuconfig aktiviert ist, wird die option bei "make" automatisch deaktiviert wenn/falls wsl erkannt wird

**Reply by fda77** [reply]
also das ist kein "bugfix" sondern ein "workaround". es müsste jemand mit wsl schauen warum ccache "nicht will"

**Reply by Conan179** [reply]
Ich hab, per google versucht,mir zu erklären, warum ccache den Ordner will /run/user/1000/ccache-tmp .
Hab leider nichts gefunden :(

**Reply by aus-sua** [reply]
Ich habe einmal die Fehlermeldung (/run/user/1000/ccache-tmp) in google eingeworfen, da
sind innerhalb der ersten Suchergebnisse:
--> https://github.com/ccache/ccache/issues/703
was wohl im Endeffekt auch in einer Deaktivierung des ccache in Sandboxen endete ;-)
und
--> https://github.com/vn971/rua/blob/master/docs/ccache.md
in dem wohl der Context zur Fehlermeldung in Sandboxen beschrieben ist
und wohl als beste Lösung ein seperates Temp-Verzeichnis für ccache in Sandboxen
per script-ccache-Aufruf empfohlen wird
Aber: ich habe auch keinerlei "skills" zu wsl/Sandboxen + ccache

**Reply by fda77** [reply]
Danke für die links @aus-sua , ich hatte nichts dazu gefunden.
Normalerweise gibt es den ccache vom System, der auch eine Konfiguration entsprechend hat.
Freetz macht dazu einen ccache für den kernel und einen für die libs+binaries - diese benutzen die Config der Systemes mit! 

Daher sieht es für mich so aus als oby die BS-config "ab Werk" bei WSL kaputt ist. Vermutlich kann man dann auch sonst nichts compilieren. Und es sollte aktuell okay sein ccache bei WSL zu deaktivieren

Das zeigt alles von meinem System + Freetz + andere Crosscompiler
```
$ ccache -s
Cacheable calls:   4982951 / 7954529 (62.64%)
  Hits:            3215032 / 4982951 (64.52%)
    Direct:        2682363 / 3215032 (83.43%)
    Preprocessed:   532669 / 3215032 (16.57%)
  Misses:          1767919 / 4982951 (35.48%)
Uncacheable calls: 2969445 / 7954529 (37.33%)
Errors:               2133 / 7954529 ( 0.03%)
```
